vdr/vdr.go

```diff
@@ -147,7 +147,7 @@ func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, _ *t
 		Hash:             &current,
 		AllowDeactivated: true,
 	}
-	currentDIDDocument, _, err := r.store.Resolve(id, resolverMetadata)
+	currentDIDDocument, currentMeta, err := r.store.Resolve(id, resolverMetadata)
```

Comment:
I have mixed feelings about this; updating one's DID Document automatically "fixes" any conflicts because it refers to all source transactions, whether that's a single one or multiple.

Should we protect administrators from accidentally overwriting a conflicting change, since the new document might not include changes from both/all documents since the administrator didn't know about it at time of drawing up the document? For instance by checking the number of source transactions, requiring a flag to be passed when there's multiple?

If it's too intrusive here, we could add it just in the REST API and CLI.

Reply:
That's why we introduced a `hash` in the API that must match the current state. That way we're sure the admin has seen the current state.